src/pages/index.js

```diff
@@ -148,23 +226,123 @@ function handleEscapeKey(event) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  // change text content to "saving..."
+  const submitBtn = evt.submitter;
+  // submitBtn.textContent = "saving...";
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      // Todo - Use data argument instead of the input values
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // TODO Call setButtonText instead
+      submitBtn.textContent = "save";
+    });
 }
 
+// TODO - implement loading text for all other form submissions
+
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
+  const submitBtn = evt.submitter;
 
-  console.log(cardNameInput);
-  console.log(cardLinkInput);
+  setButtonText(submitBtn, true, "save", "saving...");
 
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
-  evt.target.reset(settings);
-  disableButton(cardSubmitBtn, settings);
-  closeModal(cardModal);
+  api
+    .addCard(inputValues)
+    .then((res) => {
+      // renderCard(res, "prepend"); // optional
+
+      const cardEl = getCardElement(res);
+      cardsList.prepend(cardEl);
+      evt.target.reset(settings);
+      disableButton(cardSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "save", "saving...");
+    });
+}
+
+// TODO - FINISH avatar submission handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const newAvatarUrl = avatarInput.value;
+
+  // TODO - prevent behavior
+  // TODO - Call api.editAvatarUserInfo
+
+  if (!newAvatarUrl) {
+    return;
+  }
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editAvatarInfo(newAvatarUrl)
+    .then((data) => {
+      console.log(data.avatar);
+      // make this work- add the src request
+      const profileAvatar = document.querySelector(".profile__avatar");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine